packages/controller/src/resources/monitoring/tenants/grafanaDatasources.ts

```diff
@@ -255,43 +255,39 @@ http {
               ${generalProxyConfig}
 
               resolver ${dnsResolver};
-              set $backend http://querier.loki.svc.cluster.local:1080/$request_uri;
+              set $backend http://querier.loki.svc.cluster.local:1080;
               proxy_pass $backend;
             }
 
             location /prometheus/api/v1/rules {
               ${generalProxyConfig}
 
-              # Check https://github.com/opstrace/opstrace/issues/896 for more details.
               resolver ${dnsResolver};
-              set $backend http://ruler.loki.svc.cluster.local:1080/prometheus/api/v1/rules$request_uri;
+              set $backend http://ruler.loki.svc.cluster.local:1080/prometheus/api/v1/rules$is_args$query_string;
```

Comment:
again, the insight here is that what's written here _fully defines_ the URL that the nginx upstream is going to see. We want to drop the `/prometheus/api/v1/rules` part of the 'original' request URL and therefore should not use `$request_uri`.